File: BOJ/lgd/1339.cpp
#include <bits/stdc++.h>

using namespace std;

/*
*/

struct alpha {
	char a;
	int value;
	int level;
};

bool sort_alpha(const alpha& s1, const alpha& s2) {
	return s1.level > s2.level;
}

int main() {
	cin.tie(nullptr);
	cout.tie(nullptr);
	ios_base::sync_with_stdio(false);
	int N;

	cin >> N;
	string input[10];
	vector<alpha> alphas;

	int result = 0;
	int start = 9;

	for (int i = 0; i < N; i++) {
		cin >> input[i];
	}
	
	
	for (int i = 0; i < N; i++) {
		for (int j = 0; j < input[i].length(); j++) {
			char temp = input[i][j];
			bool togle = true;
			int level = 1;
			for (int l = 0; l < input[i].length() - j - 1; l++) {
				level = level * 10;
			}
			for (int k = 0; k < alphas.size(); k++) {
				if (alphas[k].a == temp) {
					togle = false;		
					alphas[k].level += level;
					break;
				}
			}
			if (togle) {
				alpha t;
				t.a = temp;
				t.value = -1;
				t.level = level;
				alphas.push_back(t);
			}
		}
	}

	sort(alphas.begin(), alphas.end(), sort_alpha);


	for (int i = 0; i < alphas.size(); i++) {
		alphas[i].value = start--;
		result += alphas[i].value * alphas[i].level;
	}

	cout << result;

	return 0;
}

File: BOJ/khj/1339.py
# string to int 함수
def string_to_int(n):
    result = []
    for i in list(n):
        if "A" == i:
            result.append("9")
        elif "B" == i:
            result.append("4")
        elif "C" == i:
            result.append("8")
        elif "D" == i:
            result.append("6")
        elif "E" == i:
            result.append("5")
        elif "F" == i:
            result.append("3")
        elif "G" == i:
            result.append("7")

    return int("".join(result))

n = int(input())
result = 0
for _ in range(n):
    result += string_to_int(input())

print(result)